paymentsheet/src/main/java/com/stripe/android/paymentsheet/ui/EditCardDetailsInteractor.kt

```diff
@@ -0,0 +1,95 @@
+package com.stripe.android.paymentsheet.ui
+
+import androidx.compose.runtime.Immutable
+import com.stripe.android.CardBrandFilter
+import com.stripe.android.model.CardBrand
+import com.stripe.android.model.PaymentMethod
+import com.stripe.android.paymentsheet.CardUpdateParams
+import kotlinx.coroutines.flow.StateFlow
+
+internal typealias CardDetailsCallback = (CardUpdateParams?) -> Unit
+
+internal typealias BrandChoiceCallback = (CardBrand) -> Unit
+
+/**
+ * Interface for handling UI interactions when editing card details.
+ */
+internal interface EditCardDetailsInteractor {
+    /**
+     * The card being edited.
+     */
+    val card: PaymentMethod.Card
+
+    /**
+     * Filter for determining which card brands are available.
+     */
+    val cardBrandFilter: CardBrandFilter
+
+    /**
+     * Icon resource ID for the payment method.
+     */
+    val paymentMethodIcon: Int
+
+    /**
+     * Whether to show the card brand dropdown.
+     */
+    val showCardBrandDropdown: Boolean
+
+    /**
+     * Current state of the card edit UI.
+     */
+    val state: StateFlow<State>
+
+    /**
+     * Callback for when the card brand choice changes.
+     */
+    val onBrandChoiceChanged: BrandChoiceCallback
+
+    /**
+     * Callback for when card details change. It provides the values needed to
+     * update the card, if any.
+     */
+    val onCardDetailsChanged: CardDetailsCallback
```

Comment:
Imho, these comments hurt more than the help -- imho, i'd remove them because the variable names seem to speak for themselves, but i'll leave it up to you guys.